Mixpanel/Mixpanel.m

```diff
@@ -278,8 +278,13 @@ - (void)identify:(NSString *)distinctId
     }
     
     dispatch_async(self.serialQueue, ^{
-        self.distinctId = distinctId;
-        self.people.distinctId = distinctId;
+        // identify only changes the distinct id if it doesn't match either the existing or the alias;
+        // if it's new, blow away the alias as well.
+        if (distinctId != self.distinctId && distinctId != self.alias) {
+            self.alias = nil;
+            self.distinctId = distinctId;
+            self.people.distinctId = distinctId;
+        }
         if (self.people.unidentifiedQueue.count > 0) {
             for (NSMutableDictionary *r in self.people.unidentifiedQueue) {
                 r[@"$distinct_id"] = distinctId;
```

Comment:
i haven't looked into this a ton but at first glance it's changing a bunch of `$distinct_id` to something that you might not be setting as your actual distinct id. `unidentifiedQueue` is filled by `addPeopleRecordToQueueWithAction:andProperties:`. do we want this to happen if `distinctId` wasn't changed? (i don't know the right answer)